### DISCUSSION TITLE
Should tabs (0x09) be destructive?

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#12112] [Q&A] [answered]
I thought this would have been discussed before, but I couldn't find anything on it.

In older consoles (my only old one is Windows 7) tabs (0x09) exhibit a "destructive" behavior; that is, old text is overwritten with spaces.  More up-to date consoles (and WT) don't behave that way.  I'm not saying which is "correct" but the older behavior seems more useful.  Here's the difference, Windows 7 on top, Windows 10 on the bottom.

![image](https://user-images.githubusercontent.com/61856645/136986456-6ab6f0d6-2eaa-45ff-959d-7addb7444ea4.png)

![image](https://user-images.githubusercontent.com/61856645/136986551-85ed734c-ff46-4b47-bc94-ad3b7dba5d17.png)

**Comment by DHowett** [comment]
In VT mode, Horizontal Tab only advances the cursor to the next tab stop.

**Comment by vefatica** [comment]
> In VT mode, Horizontal Tab only advances the cursor to the next tab stop.

Yes, that's obvious.  Is that what I'd see if I could boot up Linux to a CUI, or to some emulator in XWindows?

**Comment by DHowett** [comment]
Yep! Totally is.

**Comment by DHowett** [comment, marked as answer]
What I mean to say is:

In VT mode, Horizontal Tab _is only supposed to_ advance the cursor to the next tab stop. :smile:

**Comment by vefatica** [comment]
That's clear enough.  I didn't know what it was supposed to do.
